### PR TITLE
Create a General Endpoints Page

### LINKED ISSUE #2902: Generalize Kusama Endpoints Page
Update kusama-endpoints.md to a general page on endpoints.

See https://github.com/w3f/polkadot-wiki/pull/2878#issuecomment-997467450

### PR DESCRIPTION
**Pull request by a-jwc**
Resolves #2902

### CODE REVIEW

File: docs/build/build-node-interaction.md [hunk 1]
```diff
@@ -414,3 +414,11 @@ return an error report, e.g.:
     "cause": "Upstream error description"
 }
 ```
+
+### RPC Node APIs
```
Comment (salmad3) [inline docs/build/build-node-interaction.md]: I would suggest adding the table before the **Polkadot RPC** subheading. 
We can also add a brief explanation for each endpoint, similar to the description in *kusama-endpoints.md* for `wss://kusama-rpc.polkadot.io/`.

**Review by salmad3 — CHANGES_REQUESTED** [review]

File: docs/build/build-node-interaction.md (hunk 1 above)
Comment (a-jwc) [reply docs/build/build-node-interaction.md]: done, please review this update. thanks!

File: docs/build/build-node-interaction.md [hunk 2]
```diff
@@ -17,6 +17,30 @@ documentation for the tool you are using:
 **Polkadot-JS RPC** is a JavaScript library for interacting with the **Substrate RPC API** endpoint, distributed as `@polkadot/api` Node.js package.  
 **Substrate API Sidecar** is using the **Polkadot-JS RPC** to provide separately runnable REST services.
 
+### RPC Node Endpoints
+
+For developer convenience, [Parity Tech](https://www.parity.io/) maintains archive nodes for Polkadot, Kusama, and their test networks with public endpoints. These endpoints can be used with [Polkadot-JS API](https://polkadot.js.org/docs/api) to interact with their respective chains.
```
Comment (salmad3) [inline docs/build/build-node-interaction.md]:
```suggestion
For developer convenience, [Parity Tech](https://www.parity.io/) maintains archive nodes for Polkadot, Kusama, and their test networks with public endpoints. These endpoints can be used with the [Polkadot-JS API](https://polkadot.js.org/docs/api) to interact with their respective chains. 
```

File: docs/build/build-node-interaction.md (hunk 1 above)
Comment (salmad3) [reply docs/build/build-node-interaction.md]: Let's also run it by @joepetrowski

**Comment by joepetrowski** [comment]
This should not be part of this page.

**Comment by salmad3** [comment]
> This should not be part of this page.

@a-jwc lets update the kusama-endpoints.md to just a general doc on endpoints, called _maintain-endpoints.md_, then move the proposed content in this PR to that doc and we can use conditional rendering for wss://rpc.polkadot.io & wss://kusama-rpc.polkadot.io/ for the Polkadot Wiki and Kusama Guide, respectively. For each endpoint, we can add the JS source code used to call the endpoint, as already done for wss://rpc.polkadot.io here & wss://kusama-rpc.polkadot.io/ on kusama-endpoints.md, and finally, link to the right references for more details.

maintain-endpoints.md can live under "maintain" right after maintain-polkadot-parameters.md.

**Review by salmad3 — CHANGES_REQUESTED** [review]
Thanks @a-jwc:) 

Shouldn't we update the Kusama Guide to include the same endpoints.md page instead of the previous kusama-endpoints.md doc?

File: kusama-guide/static/css/custom.css [hunk 3]
```diff
@@ -747,3 +747,19 @@ article img {
...
+    display: block;
+    margin: 0 calc(-1 * var(--ifm-pre-padding));
+    padding: 0 var(--ifm-pre-padding);
+  }
+  
+  /* If you have a different syntax highlighting theme for dark mode. */
+  html[data-theme='dark'] .docusaurus-highlight-code-line {
+    /* Color which works with dark mode syntax highlighting theme */
+    background-color: rgb(100, 100, 100);
+  }
```
Comment (salmad3) [inline kusama-guide/static/css/custom.css]:
```suggestion
  }
  
```

File: polkadot-wiki/static/css/custom.css [hunk 4]
```diff
@@ -651,3 +651,19 @@ footer .list-social-links a {
...
+  display: block;
+  margin: 0 calc(-1 * var(--ifm-pre-padding));
+  padding: 0 var(--ifm-pre-padding);
+}
+
+/* If you have a different syntax highlighting theme for dark mode. */
+html[data-theme='dark'] .docusaurus-highlight-code-line {
+  /* Color which works with dark mode syntax highlighting theme */
+  background-color: rgb(100, 100, 100);
+}
```
Comment (salmad3) [inline polkadot-wiki/static/css/custom.css]:
```suggestion
}

```

**Review by salmad3 — APPROVED** [review]

**Pull request merged**